tools/slang-test/test-reporter.cpp

```diff
@@ -493,6 +505,8 @@ void TestReporter::addTest(const String& testName, TestResult testResult)
 
 void TestReporter::message(TestMessageType type, const String& message)
 {
+    static std::mutex mutex;
+    std::lock_guard<std::mutex> lock(mutex);
```

Comment:
Does this mutex definitely need to be process-wide, or was it just the simplest way to inject the needed synchronization?